nixos/tests/tzupdate.nix

```diff
@@ -0,0 +1,22 @@
+{ lib, ... }:
+let
+  clientNodeName = "client";
+in
+{
+  name = "tzupdate";
+
+  # TODO: Test properly:
+  # - Add server node
+  # - Add client configuration to talk to the server node
+  # - Assert that the time zone changes appropriately
+  nodes.${clientNodeName} = {
+    services.tzupdate.enable = true;
+  };
+
+  testScript = ''
+    start_all()
+    ${clientNodeName}.wait_for_unit("multi-user.target")
```

Comment:
To solve a bit the TODOs above, how about we'd `wait_for_unit('tzupdate.service')`, and then at least print:

```
timedatectl show --property=Timezone --value
```

Instead of creating a server node, I'd aim towards running this node in a well-defined proxy that would make the timezone be predictable. Do you think that'd work?

Reply:
> To solve a bit the TODOs above, how about we'd `wait_for_unit('tzupdate.service')`, and then at least print:
> 
> ```
> timedatectl show --property=Timezone --value
> ```

That would for sure be part of the test if anyone knows how to implement it. For now `wait_for_unit('tzupdate.service')` fails because it can't reach the Internet.

> Instead of creating a server node, I'd aim towards running this node in a well-defined proxy that would make the timezone be predictable. Do you think that'd work?

Whatever works. I don't have the time to dig into this any time soon, which probably means never.